lib/cluster.js

```diff
@@ -31,12 +32,16 @@ class AggregatorRegistry extends Registry {
 	 * Gets aggregated metrics for all workers. The optional callback and
 	 * returned Promise resolve with the same value; either may be used.
 	 * @param {Function?} callback (err, metrics) => any
+	 * @param {Object} opt options for
+	 * @param {Registry} opt.registry additional registry to merge with
 	 * @return {Promise<string>} Promise that resolves with the aggregated
 	 *   metrics.
 	 */
-	clusterMetrics(callback) {
+	clusterMetrics(callback, opt) {
 		const requestId = requestCtr++;
-
+		if (opt && opt.registry) {
+			additionRegistry = opt.registry;
```

Comment:
I might be missing something here as I'm new to this codebase, but it seems odd to call an instance method (`anAggregatorRegistryInstance.clusterMetrics()`) which has the side effect of affecting a static method (`AggregatorRegistry.aggregate()`). There is an implicit dependency here in that the execution path will ultimately call `AggregatorRegistry.aggregate()` with the hope the right  `additionRegistry` is still set. 

Perhaps this `additionalRegistry` should be stored with the inflight request (perhaps by modifying the shape of data stored in `requests`) state so it's metrics can later be retrieved before the call to `AggregatorRegistry.aggregate()`.

Again I'm pretty new to looking at this, so forgive me if this is off base.

Reply:
(16 months later)

Fair point, that would be a good improvement.

We already have module-level state with `AggregatorRegistry.setRegistries()`. That should maybe be changed also (in a separate PR).